### DISCUSSION TITLE
remote control

### DISCUSSION THREAD

**Discussion by juanolon** [kovidgoyal/kitty#7286] [Q&A] [answered]
Hello,
i would like to start a new instance of kitty from a script, and then execute some commands on it. something like this (not working example)
```
kitty -o allow_remote_control=yes --listen-on unix:/tmp/socket
kitty @ --to unix:/tmp/socket launch --type tab --title "server" --cwd "~/test"
kitty @ --to unix:/tmp/socket send-text --match 'title:^server' start-server
kitty @ --to unix:/tmp/socket launch --type tab --title "editor" --cwd "~/test"
kitty @ --to unix:/tmp/socket send-text --match 'title:^editor' nvim .
```
but i'm not sure if i'm on the right track. It seems that all of the opened tabs will close as soon the command is finished. also i'm not able to send the enter key.
Any help on the remote control feature of kitty? the script would be in python, it that helps..

thanks

**Comment by kovidgoyal** [comment, marked as answer]
launch --hold

and

send-key return

On Sat, Mar 30, 2024 at 03:08:39PM -0700, Juan Pablo Stumpf wrote:
> Hello,
> i would like to start a new instance of kitty from a script, and then execute some commands on it. something like this (not working example)
> ```
> kitty -o allow_remote_control=yes --listen-on unix:/tmp/socket
> kitty @ --to unix:/tmp/socket launch --type tab --title "server" --cwd "~/test"
> kitty @ --to unix:/tmp/socket send-text --match 'title:^server' start-server
> kitty @ --to unix:/tmp/socket launch --type tab --title "editor" --cwd "~/test"
> kitty @ --to unix:/tmp/socket send-text --match 'title:^editor' nvim .
> ```
> but i'm not sure if i'm on the right track. It seems that all of the opened tabs will close as soon the command is finished. also i'm not able to send the enter key.
> Any help on the remote control feature of kitty? the script would be in python, it that helps..
>
> thanks
>
> --
> Reply to this email directly or view it on GitHub:
> https://github.com/kovidgoyal/kitty/discussions/7286
> You are receiving this because you are subscribed to this thread.
>
> Message ID: ***@***.***>

--
_____________________________________

Dr. Kovid Goyal
https://www.kovidgoyal.net
https://calibre-ebook.com
_____________________________________

**Reply by juanolon** [reply]
nice! that works! it was also necessary to update kitty to the last version. follow up question: the $PATH env is not set. i'm executing the commands from python, so `--copy-env` is not working. for example when executing something like this:
`os.system(f'/usr/local/bin/kitty @ --to unix:{socket} launch --type window --cwd {path} --hold "{cmd}"')`
whereas cmd is "cd folder". i get the error: 
```
cd folder failed with error: exec: "cd folder": executable file not found in $PATH
```

**Reply by kovidgoyal** [reply]
cd is not an executable. If you want to run shell commands you run them with sh -c "{cmd}"

**Reply by juanolon** [reply]
i see. if another shell is being used, then it is also important to write the whole path to the shell. in my case `... /usr/bin/zsh -c "cmd"`. thanks!
